File: frontend/src/app/dto/imagineIfGameRound.ts
import {GameRoom} from './gameRoom';
import {Player} from './player';
import {ImagineIfQuestion} from './imagineIfQuestion';

export class ImagineIfGameRound {
  id: number;
  gameRoom: GameRoom;
  player: Player;
  round: number;
  answer: number;
  question: ImagineIfQuestion;

  constructor(args: ImagineIfGameRound = <ImagineIfGameRound>{}) {
    this.id = args.id;
    this.gameRoom = args.gameRoom;
    this.player = args.player;
    this.round = args.round;
    this.answer = args.answer;
    this.question = args.question;
  }
}

File: src/main/java/com/corelogic/schemaconverter/games/imagine/entity/ImagineIfGameRound.java
package com.corelogic.schemaconverter.games.imagine.entity;

import com.corelogic.schemaconverter.entity.GameRoom;
import com.corelogic.schemaconverter.entity.Player;
import com.corelogic.schemaconverter.entity.enums.GameRoomType;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class ImagineIfGameRound {

    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "ImagineIfGameRoundSeq")
    @SequenceGenerator(name = "ImagineIfGameRoundSeq", sequenceName = "imagine_if_game_round_id_seq", allocationSize = 1)
    @Id
    private Long id;

    @OneToOne
    private GameRoom gameRoom;

    @OneToOne
    private Player player;

    private int round;

    private int answer;

    private int selectedPlayerId;

    @OneToOne
    private ImagineIfQuestion question;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    @Transient
    private GameRoomType type = GameRoomType.IMAGINE_IF;
}

